models/model.js

```diff
@@ -1,94 +1,144 @@
+        await db.query("SELECT * FROM articles WHERE article_id = $1", [
+          articleId,
+        ])
+      ).rows.length === 0
+    ) {
+      return Promise.reject({ status: 404, msg: "Article not found" });
+    }
+    if (JSON.stringify(Object.keys(commentinfo)) !== '["username","body"]') {
+      return Promise.reject({
+        status: 400,
+        msg: "Bad Request: Incorrect format for comment",
+      });
+    }
+    if (
+      typeof commentinfo.username !== "string" ||
+      typeof commentinfo.body !== "string"
+    ) {
+      return Promise.reject({
+        status: 400,
+        msg: "Bad Request: Incorrect format for comment",
+      });
+    }
+    if (
+      (
+        await db.query("SELECT username FROM users WHERE username = $1", [
+          commentinfo.username,
+        ])
+      ).rows.length === 0
+    ) {
+      return Promise.reject({
+        status: 404,
+        msg: "username doesnt exist create a profile first before commenting",
+      });
+    }
 
-            return Promise.reject({status:400, msg: "Bad Request: Incorrect format for comment"})
-        }
-        if(typeof commentinfo.username !== 'string' || typeof commentinfo.body !== 'string'){
-           
-            return Promise.reject({status:400, msg: "Bad Request: Incorrect format for comment"})
-        }
-        if((await db.query('SELECT username FROM users WHERE username = $1',[commentinfo.username])).rows.length === 0){
-            return Promise.reject({status:404, msg: "username doesnt exist create a profile first before commenting"})
-        }
-       
-         
-    const {rows} = await db.query(`INSERT INTO comments 
+    const { rows } = await db.query(
+      `INSERT INTO comments 
     (author,body,article_id,created_at,votes)
     VALUES
     ($1,$2,$3,CURRENT_TIMESTAMP,$4)
-    RETURNING *`,[commentinfo.username,commentinfo.body,articleId,0]
-    )
-    return rows[0]
-        
+    RETURNING *`,
+      [commentinfo.username, commentinfo.body, articleId, 0]
+    );
+    return rows[0];
+  } catch (error) {}
+}
+
+async function updateArticle(adjustVotes, article_id) {
+ 
+    
+    console.log(article_id)
+    console.log(typeof adjustVotes !== 'number')
+    if ((await db.query("SELECT * FROM articles WHERE article_id = $1", [article_id,])).rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "article doesnt exist" });
     }
-    catch(error){
-      
+    if(typeof adjustVotes!== 'number'){
+        return Promise.reject({status: 400, msg: "Bad Request incorrect format"})
     }
+    let { rows } = await db.query(
+      "SELECT votes FROM articles WHERE article_id = $1",
+      [article_id]
+    );
+   const adjustedVotes = rows[0].votes+adjustVotes
+    if(adjustVotes  < 0){
+    return Promise.reject({status: 400, msg: "Bad Request votes cant go below zero"})
+    }
+    const output = await db.query(`UPDATE articles SET votes = $1 WHERE article_id = $2 RETURNING *`,[adjustedVotes,article_id])
+    
```

Comment:
Could do votes = votes +$1 , rather than updating with javascript. This would tidy this function a lot :)